src/test/java/com/salesforce/dynamodbv2/testsupport/DefaultTestSetup.java

```diff
@@ -116,24 +116,12 @@ public void setupTableData(AmazonDynamoDB amazonDynamoDb,
                                 .build()));
                     break;
                 case N:
-                    amazonDynamoDb.putItem(
-                        new PutItemRequest().withTableName(table)
-                            .withItem(ItemBuilder.builder(hashKeyAttrType, HASH_KEY_VALUE)
-                                .someField(S, SOME_OTHER_FIELD_VALUE + table + org)
-                                .rangeKey(N, RANGE_KEY_LOW_N_VALUE)
-                                .build()));
-                    amazonDynamoDb.putItem(
-                        new PutItemRequest().withTableName(table)
+                    IntStream.rangeClosed(RANGE_KEY_N_MIN, RANGE_KEY_N_MAX).forEach(i -> amazonDynamoDb
+                        .putItem(new PutItemRequest().withTableName(table)
                             .withItem(ItemBuilder.builder(hashKeyAttrType, HASH_KEY_VALUE)
                                 .someField(S, SOME_OTHER_FIELD_VALUE + table + org)
-                                .rangeKey(N, RANGE_KEY_MIDDLE_N_VALUE)
-                                .build()));
-                    amazonDynamoDb.putItem(
-                        new PutItemRequest().withTableName(table)
-                            .withItem(ItemBuilder.builder(hashKeyAttrType, HASH_KEY_VALUE)
-                                .someField(S, SOME_OTHER_FIELD_VALUE + table + org)
-                                .rangeKey(N, RANGE_KEY_HIGH_N_VALUE)
-                                .build()));
+                                .rangeKey(N, String.valueOf(i))
+                                .build())));
```

Comment:
Is this going to make our test cases a lot slower? (I generally feel torn about having the test parameter generation also initialize the fixtures.)

Reply:
are you worried about overhead like creating the stream, converting `int`s to `String`s, etc., or just that `RANGE_KEY_N_MAX - RANGE_KEY_N_MIN + 1` could be big? (here it's 10-(-1)+1 = 12—so 12 entries are added (instead of the 3 that occur in master).)

could we lazily add these elements? (I mean, tests that don't access TABLE4 don't need it populated.)

Reply:
Just that adding anything to this initialization logic multiplies by test and parameter combination. So small changes quickly balloon. Another option would be to fill the tables with more records during query test setup. If it's not too much overhead I am fine leaving it as-is; I was just curious how much slower this makes end-to-end mt-dynamo tests.